### PR TITLE
If config has fewer than 2 languages don't show language selector

### LINKED ISSUE #2919: Disable language selector screen if there's only one language
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Additional context**

**Done when**
What conditions need to be met before this issue can be closed?

### PR DESCRIPTION
**Pull request by azizova-leyla**
### Description

If config has less than 2 languages don't show language selector. 
Added some tests. Those files should be better tested, but leaving that out of this CL.

## Release notes:

If application is configured to only be available in one language through CIVIFORM_SUPPORTED_LANGUAGES don't show language selector.

### Checklist

- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)

Fixes #2919

### CODE REVIEW

File: server/app/controllers/applicant/RedirectController.java [hunk 1]
```diff
@@ -78,6 +84,13 @@ public CompletionStage<Result> programByName(Http.Request request, String progra
               }
 
               Applicant applicant = applicantFuture.join();
+              ApplicantData data = applicant.getApplicantData();
```
Comment (nbeloglazov) [inline server/app/controllers/applicant/RedirectController.java:87]: Reading this method - it's not clear why we are setting preferred locale specifically here, in "programByName" action. To be setting locale sounds like something general, applied to various "entries" to the app.

File: server/app/controllers/HomeController.java [hunk 2]
```diff
@@ -62,6 +67,12 @@ public CompletionStage<Result> index(Http.Request request) {
                 // If the applicant has not yet set their preferred language, redirect to
                 // the information controller to ask for preferred language.
                 ApplicantData data = applicant.getApplicantData();
+                if (langs.availables().size() <= 1 && !data.hasPreferredLocale()) {
+                  data.setPreferredLocale(
+                      langs.availables().isEmpty()
+                          ? Locale.US
+                          : langs.availables().get(0).toLocale());
+                }
```
Comment (nbeloglazov) [inline server/app/controllers/HomeController.java:75]: Do we have a util file where we can share "setPreferredLocale" method with the other controller?

File: server/app/controllers/HomeController.java [hunk 3]
```diff
@@ -62,6 +67,12 @@ public CompletionStage<Result> index(Http.Request request) {
                 // If the applicant has not yet set their preferred language, redirect to
                 // the information controller to ask for preferred language.
                 ApplicantData data = applicant.getApplicantData();
+                if (langs.availables().size() <= 1 && !data.hasPreferredLocale()) {
```
Comment (nbeloglazov) [inline server/app/controllers/HomeController.java:70]: Hm, I think it worth a comment here saying that we are setting preferred locale when a single language is available so that step of language selection can be automatically skipped. Otherwise it's a bit unclear why is it useful to "choose" locale for user.

File: server/test/controllers/HomeControllerAuthenticatedTest.java [hunk 4]
```diff
@@ -53,4 +62,56 @@ public void testAuthenticatedSecurePage() {
...
+            instanceOf(HttpExecutionContext.class),
+            mockLangs);
+    Result result =
+        controller.index(addCSRFToken(fakeRequest()).build()).toCompletableFuture().join();
+    assertThat(result.redirectLocation())
+        .contains(
+            controllers.applicant.routes.ApplicantProgramsController.index(applicant.id).url());
+  }
+
+  @Test
```
Comment (clouser2) [inline server/test/controllers/HomeControllerAuthenticatedTest.java:98]: I'd also test where the locale is set to something other than the single-configured language (edge case, but good to specify the behavior).

File: server/app/controllers/HomeController.java (hunk 2 above)
Comment (azizova-leyla) [reply server/app/controllers/HomeController.java]: added util file.

File: server/app/controllers/HomeController.java [hunk 5]
```diff
@@ -62,6 +67,12 @@ public CompletionStage<Result> index(Http.Request request) {
...
+                  data.setPreferredLocale(
```
[the remark this answers is not in the document]
Comment (azizova-leyla) [reply server/app/controllers/HomeController.java]: good catch. added updateApplicant in the util method.

File: server/test/controllers/HomeControllerAuthenticatedTest.java (hunk 4 above)
Comment (azizova-leyla) [reply server/test/controllers/HomeControllerAuthenticatedTest.java:102]: done.

File: server/app/controllers/HomeController.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (azizova-leyla) [reply server/app/controllers/HomeController.java]: I think it's okay to leave preferred locale as is for the person. It's highly unlikely that they would be able to set it to some other language (it would have to available at some point, and the removed).

File: server/app/controllers/applicant/RedirectController.java (hunk 1 above)
Comment (azizova-leyla) [reply server/app/controllers/applicant/RedirectController.java]: this is deep link case. applicant would get a link directly to that program and this is the method that would be called. agree that the naming is not super clear.

File: server/app/controllers/LanguageUtils.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  }
+
+  public Applicant maybeSetDefaultLocale(Applicant applicant) {
+    ApplicantData data = applicant.getApplicantData();
+    if (data.hasPreferredLocale() || langs.availables().size() > 1) {
+      return applicant;
+    }
+    data.setPreferredLocale(
+        langs.availables().isEmpty() ? Locale.US : langs.availables().get(0).toLocale());
+    userRepository.updateApplicant(applicant);
```
Comment (clouser2) [inline server/app/controllers/LanguageUtils.java:29]: This returns a future. We should await its result.

File: server/app/controllers/LanguageUtils.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    this.langs = checkNotNull(langs);
+  }
+
+  public Applicant maybeSetDefaultLocale(Applicant applicant) {
+    ApplicantData data = applicant.getApplicantData();
+    if (data.hasPreferredLocale() || langs.availables().size() > 1) {
+      return applicant;
+    }
+    data.setPreferredLocale(
+        langs.availables().isEmpty() ? Locale.US : langs.availables().get(0).toLocale());
```
Comment (clouser2) [inline server/app/controllers/LanguageUtils.java:28]: Mind reusing `LocalizedStrings.DEFAULT_LOCALE` or pulling the constant elsewhere?

**Review by clouser2 — APPROVED** [review]

**Review by nbeloglazov — APPROVED** [review]
